File: src/C#/Sort/bubble_sort_2.cs
using System;
namespace BubbleSort2 {
    class Program {
        static int[] BubbleSort(int[] mas)
        {
            int temp;
            for (int i = 0; i < mas.Length - 1; i++)
            {
                for (int j = 0; j < mas.Length - i - 1; j++)
                {
                    if (mas[j + 1] > mas[j])
                    {
                        temp = mas[j + 1];
                        mas[j + 1] = mas[j];
                        mas[j] = temp;
                    }
                }
            }
            return mas;
        }
    }
}

File: src/Python/Sort/bubble_sort_2.py
# Короткая пузырьковая сортировка
def bubbleSort2(array):
    exchanges = True
    passnum = len(array) - 1
    while passnum > 0 and exchanges:
        exchanges = False
        for i in range(passnum):
            if array[i] > array[i + 1]:
                exchanges = True
                temp = array[i]
                array[i] = array[i + 1]
                array[i + 1] = temp
        passnum = passnum - 1


array = [5, 1, 4, 4, 3]
bubbleSort2(array)
print(array)

if __name__ == "__main__":
    bubbleSort2(array)
